### PR TITLE
Add macros for unaligned access

### PR DESCRIPTION
**Pull request by vineetgarc**

### CODE REVIEW

**Review by nick-knight — COMMENTED** [review]
The RISC-V ISA spec (and profiles, etc.) appear to consistently use the word "misaligned". Is there a reason this proposal uses "unaligned"?

**Comment by vineetgarc** [comment]
> The RISC-V ISA spec (and profiles, etc.) appear to consistently use the word "misaligned". Is there a reason this proposal uses "unaligned"?

No real reason, just the terminology I'm more used to from past. I can certainly respin with new verbiage.

File: riscv-c-api.md
Comment (kito-cheng) [inline riscv-c-api.md:105]: I would like to drop that since clang has different behavior here, it just accept and ignore `-mno-strict-align` and `-mstrict-align` for RISC-V clang.

Or move it become a note to say GCC is using that way to decide which macro should be defined.

**Comment by kito-cheng** [comment]
cc @asb @luismarques

File: riscv-c-api.md
Comment (vineetgarc) [reply riscv-c-api.md:105]: Sorry not clear what you mean: clang ignores either of them ? and uses a different toggle to affect misaligned access codegen or doesn't support that at all (extremely unlikely)

Comment (kito-cheng) [reply riscv-c-api.md:105]: > Sorry not clear what you mean: clang ignores either of them ? and uses a different toggle to affect misaligned access codegen or doesn't support that at all (extremely unlikely)

Yes, RISC-V clang accept that and then ignore that for now - clang did support misaligned code gen and some more are coming up[1], option handling in clang part can easily be implement/fix later.

Anyway just second thought about this: This should not major blocker issue here, just a minor implementation issue, and you already mention `(but not necessarily)`.

**Review by kito-cheng — COMMENTED** [review]
LGTM as a RISC-V GCC maintainer, but I would like to wait LLVM folks ack here.

**Comment by vineetgarc** [comment]
> I think from #32, @aswaterman was concerned that `_bad` was unnecessarily perjorative and `_avoid` might be preferable.

Fair enough. Updated pull request with bad -> avoid
 
> I had a slight preference for zicclsm even if it's not very comprehensible just to avoid introducing too many ways of saying the same thing.

I tend to disagree still, but we can agree to disagree :-)

> But I don't feel strongly enough to block this. I think we should wait for @aswaterman's feedback though, as he participated in the other thread.

Sure thing.

**Review by kito-cheng — APPROVED** [review]
LGTM, and I am OK with those name

ping  @aswaterman

**Comment by aswaterman** [comment]
I'm OK with this scheme.  I guess I agree with @asb in principle, but, like him, I don't think it's actually a problem, so I don't want to hold this up.

**Comment by asb** [comment]
Thanks Andrew. Sounds like we have consensus on this then? LGTM.

**Comment by kito-cheng** [comment]
Thanks @vineetgarc , gonna merge this :)

**Pull request merged**
